tests/test_server.py

```diff
@@ -118,43 +226,143 @@ def test_get_device(self, client):
         client.delete(f'/devices/{id}')
 
     @pytest.mark.dependency(depends=['test_add_device', 'test_delete_device'])
-    def test_dpi(self, client, dpi_id=0):
-        id = self.add_device(client)
+    def test_get_dpi(self, client, dpi_id=0):
+        dpi = 800
+
+        id = self.add_device(client, {
+            'profiles': [
+                {
+                    'is_active': True,
+                    'resolutions': [
+                        {
+                            'is_active': True,
+                            'xres': dpi,
+                            'dpi_min': 100,
+                            'dpi_max': 16000
+                        }
+                    ]
+                }
+            ]
+        })
+        response = client.get(f'/devices/{id}/phys_props/dpi/{dpi_id}')
+        assert response.status_code == 200
+        assert response.json() == dpi
 
-        data = 6666
-        response = client.put(f'/devices/{id}/phys_props/dpi/{dpi_id}', json=data)
+        client.delete(f'/devices/{id}')
+
+    @pytest.mark.dependency(depends=['test_add_device', 'test_delete_device', 'test_get_dpi'])
+    def test_set_dpi(self, client, dpi_id=0):
+        id = self.add_device(client, {
+            'profiles': [
+                {
+                    'is_active': True,
+                    'resolutions': [
+                        {
+                            'is_active': True,
+                            'xres': 800,
+                            'dpi_min': 100,
+                            'dpi_max': 16000
+                        }
+                    ]
+                }
+            ]
+        })
+
+        new_dpi = 6666
+        response = client.put(f'/devices/{id}/phys_props/dpi/{dpi_id}', json=new_dpi)
+        assert response.status_code == 200
```

Comment:
The 2 tests above assess that the server API is correct.
This is good but we also need to assess that the *emulator* works. So we need the evdev node parsing and whatnot

Reply:
Sure.

Reply:
Please add a `TODO` then